toonz/sources/toonzlib/tstageobjectspline.cpp

```diff
@@ -214,10 +288,15 @@ void TStageObjectSpline::loadData(TIStream &is) {
 
 void TStageObjectSpline::saveData(TOStream &os) {
   const TStroke *stroke = getStroke();
+  QList<TPointD> interpStroke = getInterpolationStroke(); // Needed for T2D compatibility 
   os.child("splineId") << (int)m_id;
   if (!m_name.empty()) os.child("name") << m_name;
   os.child("isOpened") << (int)m_isOpened;
   os.child("pos") << m_dagNodePos.x << m_dagNodePos.y;
+  os.child("color") << (int)m_color; // Needed for T2D compatibility 
+  os.child("active") << (int)m_active; // Needed for T2D compatibility 
+  os.child("steps") << (int)m_steps; // Needed for T2D compatibility 
+  os.child("width") << (int)m_width; // Needed for T2D compatibility 
```

Comment:
You load these parameters to OT instead of just ignoring them. Is it because, there may be a case that the scene saved with T2D is opened and overwritten by OT, and then loaded by T2D again and you still need to preserve these data, right?

Even if that is the case, how about saving these tags to the file only when the values are different from the default ones? Which means that these tags won't be saved unless you are using only OT.

Reply:
> You load these parameters to OT instead of just ignoring them. Is it because, there may be a case that the scene saved with T2D is opened and overwritten by OT, and then loaded by T2D again and you still need to preserve these data, right?

Yes.  Users may be working together using either application, or may be doing something for a scene in T2D and then transfer to OT for other purposes and may want to come back. Without this, information is lost if it comes back to T2D.

> Even if that is the case, how about saving these tags to the file only when the values are different from the default ones? Which means that these tags won't be saved as long as you are using only OT.

That's fair. T2D defaults this information if not found in the file so OT doesn't need to write it if it doesn't need to.  I'll make a change that if it reads it in, it will write it back out, but if it was not read in, it won't write anything out.